### PR TITLE
Various minor cleanups

### PR DESCRIPTION
**Pull request by aruediger**
Used this script to get a list of references to closed tickets: https://gist.github.com/2beaucoup/2c860b012d48043fa329cd2cbea08a08

### CODE REVIEW

**Comment by ktoso** [comment]
OK TO TEST

**Comment by ktoso** [comment]
Great, thanks! Will review shortly.

**Comment by johanandren** [comment]
```
[error]  * class akka.http.impl.engine.TokenSourceActor does not have a correspondent in current version
[error]    filter with: ProblemFilters.exclude[MissingClassProblem]("akka.http.impl.engine.TokenSourceActor")
[error]  * object akka.http.impl.engine.TokenSourceActor does not have a correspondent in current version
[error]    filter with: ProblemFilters.exclude[MissingClassProblem]("akka.http.impl.engine.TokenSourceActor$")
[error]  * method apply(Int,akka.stream.scaladsl.Flow,java.net.InetSocketAddress,akka.http.scaladsl.settings.ConnectionPoolSettings,akka.actor.ActorSystem,akka.stream.Materializer)akka.stream.Graph in object akka.http.impl.engine.client.PoolSlot does not have a correspondent in current version
[error]    filter with: ProblemFilters.exclude[DirectMissingMethodProblem]("akka.http.impl.engine.client.PoolSlot.apply")
[error]  * method apply(akka.stream.scaladsl.Flow,java.net.InetSocketAddress,akka.http.scaladsl.settings.ConnectionPoolSettings,akka.event.LoggingAdapter,akka.actor.ActorSystem,akka.stream.Materializer)akka.stream.scaladsl.Flow in object akka.http.impl.engine.client.PoolFlow does not have a correspondent in current version
[error]    filter with: ProblemFilters.exclude[DirectMissingMethodProblem]("akka.http.impl.engine.client.PoolFlow.apply")
```

**Comment by 2beaucoup** [comment]
Thx @johanandren. Should be fixed.

File: akka-http-core/src/main/scala/akka/http/impl/engine/client/PoolFlow.scala [hunk 1]
```diff
@@ -69,16 +68,16 @@ private object PoolFlow {
 
   */
   def apply(connectionFlow: Flow[HttpRequest, HttpResponse, Future[Http.OutgoingConnection]],
-            remoteAddress: InetSocketAddress, settings: ConnectionPoolSettings, log: LoggingAdapter)(
+            settings: ConnectionPoolSettings, log: LoggingAdapter)(
```
Comment (johanandren) [inline akka-http-core/src/main/scala/akka/http/impl/engine/client/PoolFlow.scala]: remoteAddress wasn't used?

File: akka-http-core/src/main/scala/akka/http/impl/engine/client/PoolSlot.scala [hunk 2]
```diff
@@ -50,7 +48,6 @@ private object PoolSlot {
                                                                                                  v
    */
   def apply(slotIx: Int, connectionFlow: Flow[HttpRequest, HttpResponse, Any],
-            remoteAddress: InetSocketAddress, // TODO: remove after #16168 is cleared
```
Comment (2beaucoup) [inline akka-http-core/src/main/scala/akka/http/impl/engine/client/PoolSlot.scala]: @johanandren I mechanically followed upwards from here.

File: akka-http-core/src/main/scala/akka/http/impl/engine/client/PoolSlot.scala [hunk 3]
```diff
@@ -63,7 +60,7 @@ private object PoolSlot {
           val actor = system.actorOf(Props(new SlotProcessor(slotIx, connectionFlow, settings)).withDeploy(Deploy.local),
             name)
           ActorProcessor[RequestContext, List[ProcessorOut]](actor)
-        }.mapConcat(conforms)
+        }.mapConcat(identity)
```
Comment (ktoso) [inline akka-http-core/src/main/scala/akka/http/impl/engine/client/PoolSlot.scala]: `ConstantFun.identity` - saves an allocation

File: akka-http-core/src/main/scala/akka/http/impl/engine/client/PoolSlot.scala (hunk 2 above)
Comment (ktoso) [inline akka-http-core/src/main/scala/akka/http/impl/engine/client/PoolSlot.scala]: Wow 16168 was ancient :)

File: akka-http-core/src/main/scala/akka/http/impl/engine/TokenSourceActor.scala [hunk 4]
```diff
@@ -1,40 +0,0 @@
...
-    if (triggered > 0 && totalDemand > 0) {
-      onNext(token)
-      triggered -= 1
-      tryDispatch()
-    }
-}
-
-private[engine] object TokenSourceActor {
-  case object Trigger
-}
```
Comment (ktoso) [inline akka-http-core/src/main/scala/akka/http/impl/engine/TokenSourceActor.scala]: yay

File: akka-http-core/src/main/scala/akka/http/impl/engine/ws/Handshake.scala [hunk 5]
```diff
@@ -80,7 +80,7 @@ private[http] object Handshake {
       val protocol = find[`Sec-WebSocket-Protocol`]
       val clientSupportedSubprotocols = protocol.toList.flatMap(_.protocols)
       // Extension support is optional in WS and currently unsupported.
-      // FIXME See #18709
+      // TODO See #18709
```
Comment (ktoso) [inline akka-http-core/src/main/scala/akka/http/impl/engine/ws/Handshake.scala]: yeap, thanks

File: akka-http-core/src/main/scala/akka/http/impl/engine/client/PoolSlot.scala [hunk 6]
```diff
@@ -183,7 +180,7 @@ private object PoolSlot {
         } else {
           inflightRequests.map { rc ⇒
             if (rc.retriesLeft == 0) {
-              val reason = error.fold[Throwable](new UnexpectedDisconnectException("Unexpected disconnect"))(conforms)
+              val reason = error.fold[Throwable](new UnexpectedDisconnectException("Unexpected disconnect"))(identity)
```
Comment (ktoso) [inline akka-http-core/src/main/scala/akka/http/impl/engine/client/PoolSlot.scala]: `ConstantFun.identity`

File: project/MiMa.scala [hunk 7]
```diff
@@ -763,7 +763,7 @@ object MiMa extends AutoPlugin {
 
         // internal api
         FilterAnyProblemStartingWith("akka.stream.impl"),
-        FilterAnyProblemStartingWith("akka.http.impl.engine.parsing.BodyPartParser"),
+        FilterAnyProblemStartingWith("akka.http.impl"),
```
Comment (ktoso) [inline project/MiMa.scala]: 👍

**Comment by ktoso** [comment]
LGTM, thanks

File: akka-http-core/src/main/scala/akka/http/impl/engine/client/PoolSlot.scala (hunk 6 above)
Comment (ktoso) [inline akka-http-core/src/main/scala/akka/http/impl/engine/client/PoolSlot.scala]: I'll skim the codebase later to replace all of those to the constant one

**Comment by 2beaucoup** [comment]
Thx @ktoso! What about `ConstantFun`?

**Comment by ktoso** [comment]
If you have a second would be cool to do a sweeping change from the scala identity to ConstantFun across the codebase :)

**Comment by 2beaucoup** [comment]
Will look into it.

**Comment by ktoso** [comment]
Thanks!

-- 
Konrad `ktoso` Malawski
Akka @ Lightbend

On 6 May 2016 at 10:53:20, 2beaucoup (notifications@github.com) wrote:

Will look into it.

**Comment by drewhk** [comment]
LGTM

**Pull request merged**